bot/src/keeper.ts

```diff
@@ -93,38 +89,16 @@ const checkAndParticipateIfPossible = async function (auction: AuctionInitialInf
 
     // try to authorize the wallet then return
     if (!isWalletAuth) {
-        console.info(`keeper: wallet "${walletAddress}" has not been authorized yet. Attempting authorization now...`);
-        const transactionHash = await authorizeWallet(ETHEREUM_NETWORK, walletAddress, false);
-        console.info(`keeper: wallet "${walletAddress}" successfully authorized via "${transactionHash}" transaction`);
-        await checkAndParticipateIfPossible(auction);
-        return;
+        await authorizeKeeperWallet(async () => await checkAndParticipateIfPossible(auction));
```

Comment:
Why do you need to check `isWalletAuth` here if you will check it within the function? Should function then be called `checkAnd...`?

Reply:
Removed this!